### PR TITLE
Add script to list failed cross version tests

### PR DESCRIPTION
**Pull request by harupy**
Signed-off-by: harupy <hkawamura0130@gmail.com>

## What changes are proposed in this pull request?

Add a python script to list failed cross version tests. This script will be called by our internal Jenkins job that automatically files ES tickets.

## How is this patch tested?

Unit tests

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: dev/list_failed_cross_version_tests.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+def main():
+    session = create_session()
+    url = (
+        "https://api.github.com/repos/mlflow/mlflow/actions/workflows/cross-version-tests.yml/runs"
+    )
+    resp = session.get(url, params={"event": "schedule"})
+    latest_run_id = resp.json()["workflow_runs"][0]["id"]
+    failed_jobs = fetch_failed_jobs(session, latest_run_id)
+    print(json.dumps(failed_jobs, indent=2))
```
Comment (BenWilson2) [inline dev/list_failed_cross_version_tests.py:57]: should the session be closed at the end to free the tcp connection? (might be a bit of a nit here)
Alternatively to manually specifying a close on the session, the `.get()`'s could be wrapped in a with clause at which the pool will be gc'd and the threads reallocated.
By default requests.Session uses urllib3's keep-alive setting == True. urllib3 will release a connection once the entire body has been read for pool reuse, but I've always just been in the habit of forcibly terminating a Session for fear of a rogue process just hanging around in a machine.

File: dev/list_failed_cross_version_tests.py [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        resp = super().request(*args, **kwargs)
+        resp.raise_for_status()
+        return resp
+
+
+def create_session():
+    session = StrictSession()
+    token = os.environ.get("GITHUB_TOKEN")
+    if token:
+        session.headers.update({"Authorization": f"token {token}"})
```
Comment (BenWilson2) [inline dev/list_failed_cross_version_tests.py:45]: this throws if someone doesn't have a token with authorization to get job results, right?

Comment (harupy) [reply dev/list_failed_cross_version_tests.py:45]: You can run this script with a token, but you will hit the rate limit pretty quickly:

https://docs.github.com/en/rest/overview/resources-in-the-rest-api#rate-limiting

> When using the built-in GITHUB_TOKEN in GitHub Actions, the rate limit is 1,000 requests per hour per repository. For organizations that belong to a GitHub Enterprise Cloud account, this limit is 15,000 requests per hour per repository.

> For unauthenticated requests, the rate limit allows for up to 60 requests per hour. Unauthenticated requests are associated with the originating IP address, and not the user making requests.

File: dev/list_failed_cross_version_tests.py (hunk 1 above)
Comment (harupy) [reply dev/list_failed_cross_version_tests.py:57]: Good point, let's use `with`.

**Review by BenWilson2 — APPROVED** [review]
LGTM

**Comment by harupy** [comment]
I don't think we need this script anymore.

**Pull request closed without merging**